gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/util/ObservationUtilTest.java

```diff
@@ -0,0 +1,289 @@
+    private static final String XML_RESOURCES_BASE = "xml/Observation/";
+    private static final String ISSUED_EHR_COMPOSITION_EXAMPLE = "2020-01-01T01:01:01.000+00:00";
+    private static final String ISSUED_EHR_EXTRACT_EXAMPLE = "2020-02-01T01:01:01.000+00:00";
+    private static final String INTERPRETATION_SYSTEM = "http://hl7.org/fhir/v2/0078";
+    private static final String EFFECTIVE_START_DATE_1 = "2010-05-21";
+    private static final String EFFECTIVE_START_DATE_2 = "2010-05-20";
+    private static final String EFFECTIVE_END_DATE = "2010-05-22";
+    private static final String PQ_QUANTITY_VALUE = "100";
+    private static final String IVL_PQ_QUANTITY_VALUE = "200";
+    private static final String QUANTITY_UNIT = "ml";
+    private static final String QUANTITY_EXTENSION_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
+        + "-ValueApproximation-1";
+
+    @SneakyThrows
+    private RCMRMT030101UK04EhrExtract unmarshallEhrExtractElement(String fileName) {
+        return unmarshallFile(getFile("classpath:" + XML_RESOURCES_BASE + fileName), RCMRMT030101UK04EhrExtract.class);
+    }
+
+    private RCMRMT030101UK04ObservationStatement getObservationStatementFromEhrExtract(RCMRMT030101UK04EhrExtract ehrExtract) {
+        return ehrExtract.getComponent().get(0).getEhrFolder().getComponent().get(0).getEhrComposition().getComponent().get(0)
+            .getObservationStatement();
+    }
+
+    private void assertInterpretation(CodeableConcept interpretation, String text, String code, String display) {
+        assertThat(interpretation.getText()).isEqualTo(text);
+        assertThat(interpretation.getCoding().get(0).getCode()).isEqualTo(code);
+        assertThat(interpretation.getCoding().get(0).getDisplay()).isEqualTo(display);
+        assertThat(interpretation.getCoding().get(0).getSystem()).isEqualTo(INTERPRETATION_SYSTEM);
+    }
+
+    @Test
+    public void mapValueQuantityUsingPqQuantity() {
+        var ehrExtract = unmarshallEhrExtractElement(
+            "pq_value_quantity_observation_example.xml");
+        var observationStatement = getObservationStatementFromEhrExtract(ehrExtract);
+
+        var quantity = ObservationUtil.getValueQuantity(observationStatement.getValue(),
+            observationStatement.getUncertaintyCode());
+
+        assertThat(quantity.getValue().toString()).isEqualTo(PQ_QUANTITY_VALUE);
+        assertThat(quantity.getUnit()).isEqualTo(QUANTITY_UNIT);
+    }
+
+    @Test
+    public void mapValueQuantityUsingIvlPqQuantity() {
+        var ehrExtract = unmarshallEhrExtractElement(
+            "ivl_pq_value_quantity_observation_example.xml");
+        var observationStatement = getObservationStatementFromEhrExtract(ehrExtract);
+
+        var quantity = ObservationUtil.getValueQuantity(observationStatement.getValue(),
+            observationStatement.getUncertaintyCode());
+
+        assertThat(quantity.getValue().toString()).isEqualTo(IVL_PQ_QUANTITY_VALUE);
```

Comment:
You could compare BigDecimals directly, having IVL_PQ_QUANTITY_VALUE = new BigDecimal(200)
```suggestion
        assertThat(quantity.getValue()).isEqualTo(IVL_PQ_QUANTITY_VALUE);
```
(in other places as well please)